src/orange/OrangeTrackView.hh

```diff
@@ -408,8 +523,10 @@ CELER_FUNCTION void OrangeTrackView::move_internal(real_type dist)
  */
 CELER_FUNCTION void OrangeTrackView::move_internal(Real3 const& pos)
 {
-    states_.pos[thread_] = pos;
-    states_.surf[thread_] = {};
+    LevelStateAccessor lsa(&states_, thread_, states_.level[thread_]);
```

Comment:
Maybe add a second constructor to the LSA that takes just the first two arguments, and dispatches by accessing the level of the current thread:
```c++
LevelStateAccessor(const StateRef* states, ThreadId tid)
    : LevelStateAccessor(states, tid, states->level[tid])
{}
```